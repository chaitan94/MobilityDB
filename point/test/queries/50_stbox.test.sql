-------------------------------------------------------------------------------
-- STbox
-------------------------------------------------------------------------------

SELECT stbox 'STBOX((1.0, 2.0), (1.0, 2.0))';
SELECT stbox 'STBOX Z((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
SELECT stbox 'STBOX T((1.0, 2.0, 2001-01-03), (1.0, 2.0, 2001-01-03))';
SELECT stbox 'STBOX ZT((1.0, 2.0, 3.0, 2001-01-04), (1.0, 2.0, 3.0, 2001-01-04))';
SELECT stbox 'GEODSTBOX((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
SELECT stbox 'GEODSTBOX T((1.0, 2.0, 3.0, 2001-01-04), (1.0, 2.0, 3.0, 2001-01-03))';
SELECT stbox 'STBOX T(( , , 2001-01-03), ( , , 2001-01-03))';

SELECT stbox 'STBOX ZT((5,6,7,2001-01-08), (1,2,3,2001-01-04))';

/* Errors */
SELECT stbox 'AAA(1, 2, 3)';
SELECT stbox 'stbox(1, 2, 3)';
SELECT stbox 'stbox((AA, 2, 3))';
SELECT stbox 'stbox((1, AA, 3))';
SELECT stbox 'stbox z((1, 2, AA))';
SELECT stbox 'stbox t((1, 2, AA))';
SELECT stbox 'stbox((1, 2, 3))';
SELECT stbox 'stbox t((1, 2, 2001-01-03))';
SELECT stbox 'stbox t((1, 2, 2001-01-03),()';
SELECT stbox 'stbox t((1, 2, 2001-01-03),(1)'; 
SELECT stbox 'stbox z((1, 2, 3),(1,2)'; 
SELECT stbox 'stbox t((1, 2, 2001-01-03),(1,2)'; 
SELECT stbox 'stbox t((1, 2, 2001-01-03),(1,2,2001-01-03)'; 

SELECT stbox(1,2,3,4);
SELECT stbox(1,2,3,4,5,6);
SELECT stbox(1,2,3,'2001-01-04',5,6,7,'2001-01-08');
SELECT stboxt(1,2,'2001-01-03',4,5,'2001-01-06');
SELECT geodstbox(1,2,3,4,5,6);
SELECT geodstbox(1,2,3,'2001-01-04',5,6,7,'2001-01-08');

SELECT stbox(8,7,6,'2001-01-05',4,3,2,'2001-01-01');
SELECT stboxt(6,5,'2001-01-04',3,2,'2001-01-01');
SELECT geodstbox(8,7,6,'2001-01-05',4,3,2,'2001-01-01');

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((2,2,3,2001-01-04), (2,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((2,2,3,2001-01-04), (2,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,3,3,2001-01-04), (1,3,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,3,3,2001-01-04), (1,3,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,4,2001-01-04), (1,2,4,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,4,2001-01-04), (1,2,4,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-05), (1,2,3,2001-01-05))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-05), (1,2,3,2001-01-05))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (2,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (2,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,3,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,3,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,4,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,4,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-05))'); 
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-05))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))');
SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))', stbox 'STBOX ZT((1,2,3,2001-01-04), (1,2,3,2001-01-04))');

-------------------------------------------------------------------------------

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b = t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b <> t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b < t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b <= t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b > t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b >= t2.b;

SELECT count(*) FROM tbl_tgeompoint WHERE temp::stbox IS NOT NULL;
SELECT count(*) FROM tbl_tgeogpoint WHERE temp::stbox IS NOT NULL;

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b && t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b @> t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b <@ t2.b;
SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b ~= t2.b;

-------------------------------------------------------------------------------
