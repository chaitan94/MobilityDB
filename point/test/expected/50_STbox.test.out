/*****************************************************************************
 * STbox
 *****************************************************************************/
SELECT stbox 'STBOX((1.0, 2.0), (1.0, 2.0))';
       stbox        
--------------------
 STBOX((1,2),(1,2))
(1 row)

SELECT stbox 'STBOX Z((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
          stbox           
--------------------------
 STBOX Z((1,2,3),(1,2,3))
(1 row)

SELECT stbox 'STBOX T((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
          stbox           
--------------------------
 STBOX T((1,2,3),(1,2,3))
(1 row)

SELECT stbox 'STBOX ZT((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))';
             stbox             
-------------------------------
 STBOX ZT((1,2,3,4),(1,2,3,4))
(1 row)

SELECT stbox 'GEODSTBOX((1.0, 2.0, 3.0), (1.0, 2.0, 3.0))';
           stbox            
----------------------------
 GEODSTBOX((1,2,3),(1,2,3))
(1 row)

SELECT stbox 'GEODSTBOX T((1.0, 2.0, 3.0, 4.0), (1.0, 2.0, 3.0, 4.0))';
              stbox               
----------------------------------
 GEODSTBOX T((1,2,3,4),(1,2,3,4))
(1 row)

SELECT stbox 'STBOX T(( , , 3.0), ( , , 3.0))';
        stbox         
----------------------
 STBOX T((,,3),(,,3))
(1 row)

/* Errors */
SELECT stbox 'AAA(1, 2, 3)';
ERROR:  Could not parse STBOX
LINE 2: SELECT stbox 'AAA(1, 2, 3)';
                     ^
SELECT stbox 'stbox(1, 2, 3)';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox(1, 2, 3)';
                     ^
SELECT stbox 'stbox((AA, 2, 3))';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox((AA, 2, 3))';
                     ^
SELECT stbox 'stbox((1, AA, 3))';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox((1, AA, 3))';
                     ^
SELECT stbox 'stbox z((1, 2, AA))';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox z((1, 2, AA))';
                     ^
SELECT stbox 'stbox t((1, 2, AA))';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox t((1, 2, AA))';
                     ^
SELECT stbox 'stbox((1, 2, 3))';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox((1, 2, 3))';
                     ^
SELECT stbox 'stbox t((1, 2, 3))';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox t((1, 2, 3))';
                     ^
SELECT stbox 'stbox t((1, 2, 3),()';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox t((1, 2, 3),()';
                     ^
SELECT stbox 'stbox t((1, 2, 3),(1)';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox t((1, 2, 3),(1)';
                     ^
SELECT stbox 'stbox z((1, 2, 3),(1,2)';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox z((1, 2, 3),(1,2)';
                     ^
SELECT stbox 'stbox t((1, 2, 3),(1,2)';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox t((1, 2, 3),(1,2)';
                     ^
SELECT stbox 'stbox t((1, 2, 3),(1,2,3)';
ERROR:  Could not parse STBOX
LINE 1: SELECT stbox 'stbox t((1, 2, 3),(1,2,3)';
                     ^
SELECT stbox(1,2,3,4);
       stbox        
--------------------
 STBOX((1,2),(3,4))
(1 row)

SELECT stbox(1,2,3,4,5,6);
          stbox           
--------------------------
 STBOX Z((1,2,3),(4,5,6))
(1 row)

SELECT stbox(1,2,3,4,5,6,7,8);
             stbox             
-------------------------------
 STBOX ZT((1,2,3,4),(5,6,7,8))
(1 row)

SELECT stboxt(1,2,3,4,5,6);
          stboxt          
--------------------------
 STBOX T((1,2,3),(4,5,6))
(1 row)

SELECT geodstbox(1,2,3,4,5,6);
         geodstbox          
----------------------------
 GEODSTBOX((1,3,5),(2,4,6))
(1 row)

SELECT geodstbox(1,2,3,4,5,6,7,8);
            geodstbox             
----------------------------------
 GEODSTBOX T((1,3,5,7),(2,4,6,8))
(1 row)

SELECT stbox(8,7,6,5,4,3,2,1);
             stbox             
-------------------------------
 STBOX ZT((4,3,2,1),(8,7,6,5))
(1 row)

SELECT stboxt(6,5,4,3,2,1);
          stboxt          
--------------------------
 STBOX T((3,2,1),(6,5,4))
(1 row)

SELECT geodstbox(8,7,6,5,4,3,2,1);
            geodstbox             
----------------------------------
 GEODSTBOX T((7,5,3,1),(8,6,4,2))
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((2,2,3,4), (2,2,3,4))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((2,2,3,4), (2,2,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,3,3,4), (1,3,3,4))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,3,3,4), (1,3,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,2,4,4), (1,2,4,4))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,4,4), (1,2,4,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,2,3,5), (1,2,3,5))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,5), (1,2,3,5))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,2,3,4), (2,2,3,4))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (2,2,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,3,3,4))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,3,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,4,4))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,4,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,5))');
 stbox_cmp 
-----------
        -1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,5))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         1
(1 row)

SELECT stbox_cmp(stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))', stbox 'STBOX ZT((1,2,3,4), (1,2,3,4))');
 stbox_cmp 
-----------
         0
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b = t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b <> t2.b;
 count 
-------
  9900
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b < t2.b;
 count 
-------
  4950
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b <= t2.b;
 count 
-------
  5050
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b > t2.b;
 count 
-------
  4950
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b >= t2.b;
 count 
-------
  5050
(1 row)

SELECT count(*) FROM tbl_tgeompoint WHERE temp::stbox IS NOT NULL;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_tgeogpoint WHERE temp::stbox IS NOT NULL;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b && t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b @> t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b <@ t2.b;
 count 
-------
   100
(1 row)

SELECT count(*) FROM tbl_stbox t1, tbl_stbox t2 where t1.b ~= t2.b;
 count 
-------
   100
(1 row)

/*****************************************************************************/
